README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# TPSClus

<!-- badges: start -->
<!-- badges: end -->

`TPSClus` is an R package designed to implement fuzzy clustering of
multivariate longitudinal data using tensor product smoothing splines.
The `TPSfit()` function fits the splines to three or more variables that
are measured repeatedly over time. The `cluster.fitted()` or
`cluster.coefs()` functions can then be used to identify fuzzy clusters
in the data. Lastly, the `FKM.glm()` function can be used to generate a
generalized linear model for a distal outcome using the fuzzy clusters
and other covariates as predictors. A number of plot options are
available for visualization of the data and clusters.

## Installation

You can install the development version of TPSClus like so (need to
update):

``` r
devtools::install_github("silivingston/TPSClus", dependencies = TRUE, build_vignettes = TRUE)
```

## Example

Basic example of flow using built-in simulated dataset:

Fit splines to multivariate data:

``` r
library(TPSClus)
data(TS.sim)

fitsplines2 <- TPSfit(TS.sim, vars=c("Var1", "Var2", "Var3"),
time="Time", ID="SubjectID", knots_time=c(0, 91, 182, 273, 365),
     fit_times=c(46, 91, 137, 182, 228, 273, 319))
```

Run fuzzy clustering algorithm on the smoothed data using 3 clusters:

``` r
clusters1 <- cluster.fitted(fitsplines2, k=3, m=1.3, seed=12345, RS=5, noise=TRUE)
summary(clusters1)
#> cluster.fitted(TPSdata = fitsplines2, k = 3, m = 1.3, noise = TRUE, 
#>     seed = 12345, RS = 5)
#> 
#> 150 subjects clustered into 3 clusters + noise cluster using m = 1.3 
#> Clusters based on fitted values at times 46 91 137 182 228 273 319 
#> and slopes between points
#> 
#> Cluster summary:
#>       Cl.size Min.degree Max.degree Mean.degree
#> Clus1      47      0.489          1       0.983
#> Clus2      50      0.483          1       0.980
#> Clus3      46      0.862          1       0.991
#> Noise       7      0.933          1       0.980
#> 
#> Component 'FKM_TPS' contains the fuzzy clustering details from package 'fclust'.
#> Components of 'FKM_TPS':
#>  [1] "U"         "H"         "F"         "clus"      "medoid"    "value"    
#>  [7] "criterion" "iter"      "k"         "m"         "ent"       "b"        
#> [13] "vp"        "delta"     "stand"     "Xca"       "X"         "D"        
#> [19] "call"     
#> 
#> Cluster validity indices:
#>         PC         PE        MPC        SIL      SIL.F         XB 
#> 0.92979979 0.04137373 0.89469968 0.46242525 0.48364829 0.24117979 
#> 
#> Output dataset 'FKM_TPS_U' contains degree of cluster membership and modal
#>       cluster assignment for each object.
#> Head of dataset 'FKM_TPS_U':
#>   Id2 SubjectID        Clus1        Clus2        Clus3        Noise ClusModal
#> 1   1        17 1.309679e-05 0.9999862733 8.929241e-08 5.406552e-07         2
#> 2   2        19 9.665821e-04 0.0004435711 9.972477e-01 1.342137e-03         3
#> 3   3        34 4.719129e-06 0.9999952117 1.255723e-08 5.657849e-08         2
#> 4   4        52 1.188519e-05 0.9999878294 4.352554e-08 2.418527e-07         2
#> 5   5        55 6.427492e-04 0.0003088823 9.982691e-01 7.792961e-04         3
#> 6   6        83 3.651509e-03 0.0012300893 9.802217e-01 1.489672e-02         3
```

Plot the mean cluster trajectories:

``` r
plot(clusters1, legend_label_size=10)
#> `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'
```

<img src="man/figures/README-plot-1.png" width="100%" />

Fit a GLM using the clusters and other covariates as predictors:

``` r
model <- FKM.glm(clusters1, TS.sim, y="outcome", covariates=c("x1", "x2"),
family="binomial")
summary(model)
#> Full model:
#> Formula (f1):  outcome ~ Clus2 + Clus3 + Noise + x1 + x2 
#> Family: binomial 
#> 
#> Call:
#> glm(formula = f1, family = family, data = data3)
#> 
#> Deviance Residuals: 
#>      Min        1Q    Median        3Q       Max  
#> -1.97164  -0.22557  -0.02179   0.13942   2.11563  
#> 
#> Coefficients:
#>             Estimate Std. Error z value Pr(>|z|)    
#> (Intercept) 10.42476    2.11640   4.926 8.41e-07 ***
#> Clus2        1.15006    0.86491   1.330   0.1836    
#> Clus3        2.37069    1.00594   2.357   0.0184 *  
#> Noise        3.52023    1.73903   2.024   0.0429 *  
#> x1           0.73342    0.68421   1.072   0.2838    
#> x2          -0.27902    0.05381  -5.185 2.16e-07 ***
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> 
#> (Dispersion parameter for binomial family taken to be 1)
#> 
#>     Null deviance: 201.065  on 149  degrees of freedom
#> Residual deviance:  63.838  on 144  degrees of freedom
#> AIC: 75.838
#> 
#> Number of Fisher Scoring iterations: 7
#> 
#> 
#> ANOVA chi-square p-value for significance of clusters in model:
#> 0.03360065
```

Predict the outcome for a new dataset:

``` r
data(TS.sim.new)
predicted <- predict(model, TS.sim.new)
head(predicted)
#>   SubjectID x1   x2 Id2        Clus1       Clus2        Clus3        Noise
#> 1      1626  0 23.5   1 3.263066e-06 0.999996610 1.915778e-08 1.075745e-07
#> 2      1627  0 87.2   2 1.091953e-02 0.004496758 9.803848e-01 4.198949e-03
#> 3      1629  0 59.3   3 6.915108e-02 0.045432056 8.668493e-01 1.856759e-02
#> 4      1634  0 36.3   4 2.411935e-07 0.999999754 8.120729e-10 3.633010e-09
#> 5      1641  1 49.6   5 2.365786e-01 0.119455660 1.033256e-02 6.336331e-01

NEWS.md
# TPSClus 0.0.0.9000 (development version)

* Added a `NEWS.md` file to track changes to the package.


data-raw/TS.sim.new.R
## code to prepare `TS.sim_new` dataset goes here

TS.sim.new <- structure(list(
  SubjectID = c(1626L, 1626L, 1626L, 1626L, 1626L,
                1626L, 1626L, 1626L, 1626L, 1626L, 1627L, 1627L, 1627L, 1627L,
                1627L, 1627L, 1627L, 1627L, 1627L, 1629L, 1629L, 1629L, 1629L,
                1629L, 1629L, 1634L, 1634L, 1634L, 1634L, 1634L, 1634L, 1634L,
                1634L, 1634L, 1634L, 1634L, 1634L, 1634L, 1634L, 1634L, 1634L,
                1634L, 1634L, 1641L, 1641L, 1641L, 1641L, 1641L, 1641L, 1641L,
                1641L, 1641L, 1650L, 1650L, 1650L, 1650L, 1650L, 1650L, 1652L,
                1652L, 1652L, 1652L, 1652L, 1652L, 1652L, 1652L, 1652L, 1652L,
                1652L, 1652L, 1652L, 1655L, 1655L, 1655L, 1655L, 1655L, 1655L,
                1655L, 1655L, 1655L, 1655L, 1655L, 1674L, 1674L, 1674L, 1674L,
                1674L, 1674L, 1678L, 1678L, 1678L, 1678L, 1678L, 1678L, 1678L,
                1678L, 1678L, 1678L, 1678L, 1678L, 1686L, 1686L, 1686L, 1686L,
                1686L, 1686L, 1686L, 1686L, 1686L, 1686L, 1693L, 1693L, 1693L,
                1693L, 1693L, 1693L, 1693L, 1701L, 1701L, 1701L, 1701L, 1701L,
                1701L, 1701L, 1701L, 1720L, 1720L, 1720L, 1720L, 1720L, 1720L,
                1720L, 1720L, 1720L, 1720L, 1720L, 1720L, 1720L, 1722L, 1722L,
                1722L, 1722L, 1722L, 1722L, 1722L, 1722L, 1722L, 1722L, 1722L,
                1722L, 1745L, 1745L, 1745L, 1745L, 1745L, 1745L, 1745L, 1745L,
                1745L, 1745L, 1745L, 1745L, 1745L, 1745L, 1745L, 1745L, 1745L,
                1745L, 1746L, 1746L, 1746L, 1746L, 1746L, 1746L, 1746L, 1746L,
                1746L, 1746L, 1746L, 1746L, 1772L, 1772L, 1772L, 1772L, 1772L,
                1772L, 1772L, 1772L, 1772L, 1772L, 1772L, 1772L, 1787L, 1787L,
                1787L, 1787L, 1787L, 1787L, 1787L, 1787L, 1787L, 1787L, 1787L,
                1787L, 1787L, 1787L, 1792L, 1792L, 1792L, 1792L, 1792L, 1792L,
                1792L, 1792L, 1792L, 1792L, 1792L, 1792L, 1792L, 1792L, 1861L,
                1861L, 1861L, 1861L, 1861L, 1861L, 1868L, 1868L, 1868L, 1868L,
                1868L, 1868L, 1868L, 1868L, 1868L, 1868L, 1868L, 1868L, 1868L,
                1868L, 1888L, 1888L, 1888L, 1888L, 1888L, 1888L, 1888L, 1915L,
                1915L, 1915L, 1915L, 1915L, 1915L, 1915L, 1915L, 1915L, 1915L,
                1929L, 1929L, 1929L, 1929L, 1929L, 1929L, 1929L, 1929L, 1929L,
                1929L, 1929L, 1929L, 1929L, 1929L, 1929L, 1929L, 1929L, 1929L,
                1953L, 1953L, 1953L, 1953L, 1953L, 1953L, 1953L, 1953L, 1953L,
                1953L, 1953L, 1953L, 1967L, 1967L, 1967L, 1967L, 1967L, 1967L,
                1967L, 1967L, 1967L, 1967L, 1967L, 1967L, 1967L, 1967L, 1980L,
                1980L, 1980L, 1980L, 1980L, 1980L, 1980L, 1980L, 1980L, 1980L,
                1980L, 1990L, 1990L, 1990L, 1990L, 1990L, 1990L, 1990L, 1990L,
                1990L, 1990L, 1990L, 1990L, 1990L, 1991L, 1991L, 1991L, 1991L,
                1991L, 1991L, 1991L, 1991L, 1991L),
  Time = c(0, 14, 28, 38, 109,
           230, 241, 318, 350, 364, 0, 102, 160, 164, 169, 221, 274, 361,
           364, 0, 169, 196, 262, 350, 364, 0, 41, 57, 58, 60, 62, 96, 108,
           122, 212, 256, 268, 272, 275, 279, 280, 321, 364, 0, 25, 32,
           71, 147, 198, 216, 221, 364, 0, 29, 91, 160, 353, 364, 0, 12,
           57, 59, 85, 119, 202, 234, 235, 299, 329, 357, 364, 0, 67, 76,
           152, 160, 234, 282, 321, 331, 358, 364, 0, 63, 179, 202, 216,
           364, 0, 19, 30, 42, 115, 122, 131, 207, 249, 302, 311, 364, 0,
           111, 114, 128, 140, 229, 244, 266, 347, 364, 0, 43, 64, 107,
           221, 351, 364, 0, 50, 76, 124, 232, 260, 315, 364, 0, 92, 112,
           113, 167, 231, 300, 302, 307, 320, 325, 335, 364, 0, 34, 225,
           230, 232, 264, 270, 288, 294, 335, 350, 364, 0, 101, 126, 134,
           143, 170, 177, 184, 200, 224, 237, 256, 259, 286, 323, 326, 334,
           364, 0, 13, 28, 56, 64, 78, 95, 102, 305, 330, 337, 364, 0, 3,
           7, 12, 46, 53, 60, 100, 121, 274, 309, 364, 0, 21, 87, 106, 160,
           184, 188, 222, 240, 260, 298, 353, 357, 364, 0, 7, 25, 48, 141,
           145, 199, 202, 239, 257, 260, 304, 342, 364, 0, 10, 114, 137,
           294, 364, 0, 20, 57, 77, 139, 142, 209, 229, 231, 242, 271, 333,
           339, 364, 0, 25, 73, 132, 347, 350, 364, 0, 2, 33, 137, 191,
           197, 211, 254, 284, 364, 0, 51, 61, 68, 97, 106, 127, 134, 171,
           206, 229, 264, 283, 297, 315, 322, 343, 364, 0, 18, 80, 122,
           184, 194, 247, 249, 333, 338, 345, 364, 0, 39, 49, 93, 141, 213,
           221, 235, 239, 247, 331, 337, 351, 364, 0, 17, 38, 63, 159, 210,
           258, 277, 315, 356, 364, 0, 46, 47, 79, 111, 148, 150, 164, 181,
           254, 270, 361, 364, 0, 125, 150, 206, 276, 286, 300, 314, 364
  ),
  Var1 = c(6.1, 5.3, 5.2, 6.3, 6.9, 6.9, 8.6, 6.6, 7, 9.5, 9.3,
           5.6, 3.9, 2.1, 3.8, 4.5, 10.8, 12, 13.3, 9, 4, 4.4, 9.8, 11.2,
           11.1, 6.6, 7.9, 8.2, 6.8, 7.9, 6.9, 9, 8.3, 7.2, 8.1, 8.4, 7.3,
           7.5, 9.9, 8.9, 9.2, 9.5, 8, 7.4, 38.8, 8, 7.7, 7.2, 4.7, 5, 5.7,
           9.2, 7, 7, 9.8, 7.7, 9.6, 8.7, 6.1, 5.5, 4.3, 4.8, 6.6, 21.2,
           3.7, 4.5, 3.5, 6.6, 6.4, 5.7, 7.5, 8.6, 8.1, 8.1, 6.6, 4.9, 6.8,
           6.6, 8.3, 35.7, 10.7, 11, 8, 6.8, 2, 4, 5.5, 10.8, 6, 7.1, 8.4,
           7.6, 5.7, 8.8, 7.9, 9.8, 8, 8.3, 8.9, 7.6, 6.4, 6.7, 7.6, 7.6,
           7.8, 7.9, 8.2, 8, 8.5, 9.6, 10.1, 9.7, 7.9, 4.1, 6.3, 10.5, 10.3,
           8.2, 5.2, 4.7, 1.7, 3.7, 8.3, 9.7, 8.1, 7.4, 9.5, 9.4, 7.5, 11,
           10.5, 8.2, 8.7, 8.4, 8.4, 10.7, 8.3, 7.2, 10.9, 10.7, 6.9, 6.8,
           6.6, 8, 7.2, 7.2, 9.1, 9.8, 11.5, 12.5, 6.7, 7.2, 7.8, 6.1, 8.2,
           9.2, 7.6, 8.4, 7.9, 8.7, 9.2, 8.3, 8.3, 8.4, 9, 8.9, 10.2, 6.9,
           7.5, 6.1, 7.1, 4.8, 8, 6.5, 6.9, 7.9, 5.3, 6.3, 7.6, 8.3, 6.3,
           7.5, 6, 7.7, 6.2, 5.6, 7.6, 7.3, 5.4, 6.9, 4.5, 9.7, 8.3, 7.9,
           7.9, 10.3, 10.1, 9.4, 9.3, 10.5, 10.8, 8.5, 9.6, 8.9, 10.8, 11.3,
           7.9, 8, 6.4, 8.3, 8.3, 3.9, 7.1, 1.9, 4.8, 8.1, 6.7, 7.7, 8.3,
           10.3, 6.5, 6.7, 5.6, 4.6, 5.9, 8.5, 10, 8.3, 6, 6.6, 1.8, 2,
           4, 6.6, 6, 7.6, 9.1, 10.5, 9.5, 11.3, 7.8, 9, 5.4, 5.7, 8.2,
           8.6, 11.1, 10.4, 10, 10.9, 8.2, 8.7, 6.7, 9.5, 8.2, 10.1, 12,
           10.5, 8.8, 8.6, 5.8, 6.7, 5, 2.2, 4.1, 2.9, 7.3, 7.3, 11, 10.4,
           11.6, 13.1, 15.3, 13.5, 12.8, 8.1, 6.5, 8, 6.1, 6.7, 6.2, 4.1,
           3.6, 8.9, 7.9, 6.8, 11.2, 8.8, 11.4, 10.8, 10.4, 6.8, 8.3, 6.8,
           7.1, 7.4, 7.3, 10, 10.1, 10.2, 12.2, 6.8, 5.9, 5.3, 8.9, 8.4,
           10.2, 8.1, 9.9, 8.4, 8.6, 9.7, 8.8, 9.4, 9.5, 8.7, 9.2, 7.1,
           7.9, 5.5, 6.3, 8.2, 6.5, 9.9, 9.8, 9.2, 4.3, 4.3, 5.4, 12.3,
           13.7, 11.3, 12.2, 11.1),
  Var2 = c(8.1, 11.1, 10.3, 9.9, 10.3,
           11.9, 12.2, 11.9, 12.7, 12.5, 12.1, 15.2, 11.1, 9.1, 9.9, 9.3,
           8.5, 17, 13.4, 9, 10.7, 7.7, 6.7, 13, 15.5, 11.7, 13.1, 12.4,
           13.2, 13.7, 12.4, 12.9, 10.6, 13.9, 14.5, 15.4, 14.2, 15.9, 14.1,
           12.5, 15.4, 13.3, 14.6, 6.5, 41.8, 9.1, 9.2, 11.5, 9, 7.9, 10.2,
           6.2, 9.6, 10.3, 12.1, 11.6, 11.2, 14.7, 5.2, 3.8, 8, 8.2, 5.5,
           33.6, 4.6, 4.5, 6, 3.7, 2.2, 2.8, 2.5, 9.8, 13.2, 14.9, 16.3,
           14.3, 11.7, 11.1, 10.4, 55.4, 7.7, 10.1, 5.3, 8.5, 3.5, 4.7,
           1.6, 10.4, 10.3, 10.1, 10.5, 11.3, 12, 12.7, 10.2, 10.2, 12.5,
           12.9, 13.2, 13.7, 13, 13.4, 12.5, 13.2, 14.3, 12.7, 13.7, 14.1,
           12.6, 14.3, 9.5, 11.4, 11.5, 13.5, 6.5, 15.7, 12.4, 5.5, 11.3,
           10.2, 8.7, 2, 4.7, 6.7, 13, 11.9, 13.5, 11.2, 12.8, 13.4, 13.8,
           13.6, 15.2, 14.1, 12.7, 14.3, 14.3, 13.7, 13.1, 16, 13.7, 14.4,
           14.7, 13.1, 12.2, 13.1, 11.9, 11.8, 13.6, 12.4, 8.1, 12.5, 11.8,
           12.2, 11, 12, 12.1, 12.9, 11.6, 12.7, 11.1, 12.9, 13.3, 13.3,
           15.5, 13, 13.4, 12.7, 7.4, 8.8, 8.8, 9.4, 9.3, 10.6, 10.6, 12.3,
           8.7, 5.4, 6.6, 7.4, 7.4, 8.3, 7.4, 9.7, 9.3, 7.9, 9.7, 9.6, 10.5,
           8.1, 7.1, 6, 13.2, 15, 13.9, 14, 14, 13.1, 14.8, 15.3, 16.5,
           14.2, 14, 13, 15.2, 16.1, 10.4, 10.7, 10.5, 14.7, 13.8, 14.2,
           13.5, 13, 10.6, 10.1, 10.9, 9.8, 8.4, 9.5, 5.9, 8.6, 9.5, 10.8,
           6.2, 5.6, 6.9, 8.9, 11.8, 12.7, 11.5, 9.9, 7.1, 3.4, 2.4, 4.2,
           4.4, 10.3, 10.3, 13.3, 8.5, 9.9, 11.2, 8.3, 5.5, 7.8, 5.9, 13.1,
           13.4, 15.6, 16.8, 15.1, 15.9, 15.7, 14, 13.7, 10.7, 8.9, 14.6,
           15.9, 14.8, 14.5, 14, 13.6, 14.1, 10.3, 7.2, 6.5, 8.2, 11.2,
           10.9, 11.5, 10.2, 13.6, 15.4, 11, 11.9, 13.2, 11.4, 13.7, 11.1,
           12.8, 10.6, 7.4, 9, 9.9, 9.9, 11.1, 14.1, 13.8, 16.4, 15.8, 13.2,
           13.7, 14.3, 12.6, 11.5, 10.9, 10.1, 8.7, 8.9, 10.4, 9.5, 10.2,
           10.8, 12.6, 10.5, 12.3, 11.3, 12.3, 12.8, 11.8, 11.8, 14.1, 13,
           13.2, 11.3, 11.6, 14.6, 15.1, 13.6, 10.4, 10.5, 10.2, 9.7, 12.1,
           12.5, 11.8, 8.7, 8.4, 11.9, 11.3, 11.4, 19.3),
  Var3 = c(6.8,
           6.9, 5.7, 6.4, 5.5, 6.7, 5.9, 6.3, 7.9, 4.9, 1.6, 4.3, 6.5, 4.8,
           5.9, 8.5, 7.9, 10.8, 7.5, 5.4, 8.1, 11.4, 10.7, 12.1, 11.1, 5.1,
           5.5, 5, 3.8, 4.3, 2.6, 5.6, 5.7, 5.5, 6.4, 3.5, 5.5, 5.7, 6.1,
           5.9, 7.8, 5.6, 6.4, 6.4, 27, 7, 6.2, 6.1, 7.4, 6.6, 9, 10.1,
           4.4, 5.1, 4.9, 4.6, 7, 5.2, 5.6, 3.9, 2, 6.2, 5.6, 20.3, 6.4,
           6.4, 5, 8, 9.1, 7.8, 8.5, 3.9, 4.5, 4.4, 6.3, 4.7, 5.4, 5.6,
           7.8, 46.1, 7.2, 6.5, 3.5, 5.8, 8, 8.5, 8.8, 12.7, 2.6, 4.9, 3.8,
           4.6, 7, 4.6, 6.7, 3.5, 5, 3.6, 3.8, 5.6, 4.4, 3.5, 6.1, 4.1,
           4.6, 4.9, 5.7, 6.6, 6.4, 6.2, 3.6, 2.3, 3.7, 2.6, 8.5, 9.9, 9.3,
           3.6, 6.3, 4.5, 5.5, 9.7, 10.2, 11.2, 10.5, 8.2, 6.6, 5.4, 6.7,
           8.6, 4.9, 6.7, 7.9, 5.4, 7.5, 6.2, 7.5, 8.7, 4, 5.3, 5, 5.5,
           5.5, 7.7, 6.4, 6.5, 7.5, 7.9, 7.3, 7.3, 3.7, 6.9, 5.4, 4.1, 5.7,
           4.2, 3.3, 4.6, 6.7, 6, 5.9, 6.7, 5.6, 6.5, 5.4, 4.3, 4.7, 6.3,
           5.3, 5.6, 4.9, 6.2, 5.4, 7.4, 4.3, 6.4, 7.8, 8.9, 11.1, 10.9,
           6.6, 6.1, 6.4, 7.3, 7, 4.7, 6.2, 6, 6.5, 8.9, 10.8, 10.1, 6.1,
           5.8, 5.3, 6.5, 5.8, 6.6, 5.8, 5.2, 6.8, 6, 7.1, 7.5, 6.9, 7.4,
           4.5, 6.5, 3.9, 6.2, 7.4, 6.3, 6.7, 9.3, 7.9, 8.6, 7.6, 8.6, 10.2,
           10.5, 6.2, 2.6, 4.9, 4.3, 7.1, 8.3, 3.5, 5.1, 8.1, 5.9, 9.6,
           8.5, 10.3, 9.5, 9.1, 11.2, 12.3, 12.2, 12.1, 12.7, 4.8, 7.2,
           4.2, 5.9, 8.9, 9.9, 11.7, 3.5, 3.4, 2.7, 1.6, 4.1, 5.4, 3.8,
           4.7, 5.9, 7.8, 0.7, 0.3, 2.5, 2, 2.4, 4.5, 2.6, 2.9, 4.8, 6.9,
           7.1, 7.8, 7.9, 9.9, 6.2, 7.7, 10.2, 7.8, 3.1, 4.9, 5.6, 4.7,
           5.3, 6.3, 5.3, 5.1, 8, 6, 6.8, 7.1, 5.9, 6.2, 4.9, 5.6, 4.9,
           9.2, 8.2, 8, 8.6, 7.1, 9.3, 10.3, 9.4, 8.6, 8.7, 8.2, 6.1, 6.7,
           5.7, 7.5, 7.4, 8.8, 7.7, 8.2, 7.8, 3.8, 2, 1.3, 3.7, 6, 2.4,
           4.8, 4.9, 4.3, 6.3, 6, 7.4, 8.1, 3.7, 6.6, 7, 8.3, 10.8, 11.3,
           10.9, 12.1, 12.1),
  x1 = c(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1,
         1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,
         1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,
         1, 1, 1, 1, 1, 1, 1),
  x2 = c(23.5, 23.5, 23.5, 23.5, 23.5, 23.5,
         23.5, 23.5, 23.5, 23.5, 87.2, 87.2, 87.2, 87.2, 87.2, 87.2, 87.2,
         87.2, 87.2, 59.3, 59.3, 59.3, 59.3, 59.3, 59.3, 36.3, 36.3, 36.3,
         36.3, 36.3, 36.3, 36.3, 36.3, 36.3, 36.3, 36.3, 36.3, 36.3, 36.3,
         36.3, 36.3, 36.3, 36.3, 49.6, 49.6, 49.6, 49.6, 49.6, 49.6, 49.6,
         49.6, 49.6, 34.1, 34.1, 34.1, 34.1, 34.1, 34.1, 49.9, 49.9, 49.9,
         49.9, 49.9, 49.9, 49.9, 49.9, 49.9, 49.9, 49.9, 49.9, 49.9, 73.7,
         73.7, 73.7, 73.7, 73.7, 73.7, 73.7, 73.7, 73.7, 73.7, 73.7, 72.7,
         72.7, 72.7, 72.7, 72.7, 72.7, 36.1, 36.1, 36.1, 36.1, 36.1, 36.1,
         36.1, 36.1, 36.1, 36.1, 36.1, 36.1, 41.7, 41.7, 41.7, 41.7, 41.7,
         41.7, 41.7, 41.7, 41.7, 41.7, 26.9, 26.9, 26.9, 26.9, 26.9, 26.9,
         26.9, 43.1, 43.1, 43.1, 43.1, 43.1, 43.1, 43.1, 43.1, 91.6, 91.6,
         91.6, 91.6, 91.6, 91.6, 91.6, 91.6, 91.6, 91.6, 91.6, 91.6, 91.6,
         58.6, 58.6, 58.6, 58.6, 58.6, 58.6, 58.6, 58.6, 58.6, 58.6, 58.6,
         58.6, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1,
         38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 38.1, 22.9, 22.9, 22.9,
         22.9, 22.9, 22.9, 22.9, 22.9, 22.9, 22.9, 22.9, 22.9, 113.3,
         113.3, 113.3, 113.3, 113.3, 113.3, 113.3, 113.3, 113.3, 113.3,
         113.3, 113.3, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7,
         41.7, 41.7, 41.7, 41.7, 41.7, 41.7, 63.4, 63.4, 63.4, 63.4, 63.4,
         63.4, 63.4, 63.4, 63.4, 63.4, 63.4, 63.4, 63.4, 63.4, 57.9, 57.9,
         57.9, 57.9, 57.9, 57.9, -1.7, -1.7, -1.7, -1.7, -1.7, -1.7, -1.7,
         -1.7, -1.7, -1.7, -1.7, -1.7, -1.7, -1.7, -9.4, -9.4, -9.4, -9.4,
         -9.4, -9.4, -9.4, 51.9, 51.9, 51.9, 51.9, 51.9, 51.9, 51.9, 51.9,
         51.9, 51.9, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8,
         46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 46.8, 55.8, 55.8,
         55.8, 55.8, 55.8, 55.8, 55.8, 55.8, 55.8, 55.8, 55.8, 55.8, 63.2,
         63.2, 63.2, 63.2, 63.2, 63.2, 63.2, 63.2, 63.2, 63.2, 63.2, 63.2,
         63.2, 63.2, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7, 41.7,
         41.7, 41.7, 69.7, 69.7, 69.7, 69.7, 69.7, 69.7, 69.7, 69.7, 69.7,
         69.7, 69.7, 69.7, 69.7, 20.6, 20.6, 20.6, 20.6, 20.6, 20.6, 20.6,
         20.6, 20.6)),
  class = "data.frame",
  row.names = c(71L, 72L, 73L,
                74L, 75L, 76L, 77L, 78L, 79L, 80L, 1572L, 1573L, 1574L, 1575L,
                1576L, 1577L, 1578L, 1579L, 1580L, 1795L, 1796L, 1797L, 1798L,
                1799L, 1800L, 410L, 411L, 412L, 413L, 414L, 415L, 416L, 417L,
                418L, 419L, 420L, 421L, 422L, 423L, 424L, 425L, 426L, 427L, 1540L,
                1541L, 1542L, 1543L, 1544L, 1545L, 1546L, 1547L, 1548L, 333L,
                334L, 335L, 336L, 337L, 338L, 1518L, 1519L, 1520L, 1521L, 1522L,
                1523L, 1524L, 1525L, 1526L, 1527L, 1528L, 1529L, 1530L, 1561L,
                1562L, 1563L, 1564L, 1565L, 1566L, 1567L, 1568L, 1569L, 1570L,
                1571L, 1918L, 1919L, 1920L, 1921L, 1922L, 1923L, 143L, 144L,
                145L, 146L, 147L, 148L, 149L, 150L, 151L, 152L, 153L, 154L, 452L,
                453L, 454L, 455L, 456L, 457L, 458L, 459L, 460L, 461L, 1724L,
                1725L, 1726L, 1727L, 1728L, 1729L, 1730L, 2039L, 2040L, 2041L,
                2042L, 2043L, 2044L, 2045L, 2046L, 167L, 168L, 169L, 170L, 171L,
                172L, 173L, 174L, 175L, 176L, 177L, 178L, 179L, 1298L, 1299L,
                1300L, 1301L, 1302L, 1303L, 1304L, 1305L, 1306L, 1307L, 1308L,
                1309L, 521L, 522L, 523L, 524L, 525L, 526L, 527L, 528L, 529L,
                530L, 531L, 532L, 533L, 534L, 535L, 536L, 537L, 538L, 1393L,
                1394L, 1395L, 1396L, 1397L, 1398L, 1399L, 1400L, 1401L, 1402L,
                1403L, 1404L, 1430L, 1431L, 1432L, 1433L, 1434L, 1435L, 1436L,
                1437L, 1438L, 1439L, 1440L, 1441L, 248L, 249L, 250L, 251L, 252L,
                253L, 254L, 255L, 256L, 257L, 258L, 259L, 260L, 261L, 1105L,
                1106L, 1107L, 1108L, 1109L, 1110L, 1111L, 1112L, 1113L, 1114L,
                1115L, 1116L, 1117L, 1118L, 1456L, 1457L, 1458L, 1459L, 1460L,
                1461L, 1801L, 1802L, 1803L, 1804L, 1805L, 1806L, 1807L, 1808L,
                1809L, 1810L, 1811L, 1812L, 1813L, 1814L, 1135L, 1136L, 1137L,
                1138L, 1139L, 1140L, 1141L, 929L, 930L, 931L, 932L, 933L, 934L,
                935L, 936L, 937L, 938L, 2021L, 2022L, 2023L, 2024L, 2025L, 2026L,
                2027L, 2028L, 2029L, 2030L, 2031L, 2032L, 2033L, 2034L, 2035L,
                2036L, 2037L, 2038L, 989L, 990L, 991L, 992L, 993L, 994L, 995L,
                996L, 997L, 998L, 999L, 1000L, 1379L, 1380L, 1381L, 1382L, 1383L,
                1384L, 1385L, 1386L, 1387L, 1388L, 1389L, 1390L, 1391L, 1392L,
                132L, 133L, 134L, 135L, 136L, 137L, 138L, 139L, 140L, 141L, 142L,
                1417L, 1418L, 1419L, 1420L, 1421L, 1422L, 1423L, 1424L, 1425L,
                1426L, 1427L, 1428L, 1429L, 2217L, 2218L, 2219L, 2220L, 2221L,
                2222L, 2223L, 2224L, 2225L))


usethis::use_data(TS.sim.new, overwrite = TRUE)
